pkg/virt-handler/options_test.go

```diff
@@ -0,0 +1,175 @@
+package virthandler
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+
+	"libvirt.org/go/libvirtxml"
+)
+
+const memorySize = 16256896
+const pageAmount = 1000
+const cellDistance = 10
+const unit = "KiB"
+
+var _ = Describe("Parsing VMI Options", func() {
+	Context("virt-handler VM processes VMI options during update", func() {
+		DescribeTable("it should convert api capabilities to cmdv1 topology", func(cellCount, pageCount, cpuCount int) {
+			caps := newLibvirtCapsTopology(
+				withCells(cellCount),
+				withCellMemory(memorySize, unit),
+				withCellPages(pageCount, pageAmount, unit),
+				withCellDistances(),
+				withCellCpus(cpuCount),
+			)
+			topology := caps.Host.NUMA
+			expectedTopology := capabilitiesToTopology(caps)
+
+			Expect(expectedTopology.NumaCells).To(HaveLen(cellCount))
+			for i, cell := range topology.Cells.Cells {
+				Expect(expectedTopology.NumaCells[i].Memory).To(PointTo(matchMemory(cell.Memory)))
+				Expect(expectedTopology.NumaCells[i].Pages).To(ConsistOf(matchPages(cell.PageInfo)))
+				Expect(expectedTopology.NumaCells[i].Distances).To(ConsistOf(matchDistances(cell.Distances.Siblings)))
+				Expect(expectedTopology.NumaCells[i].Cpus).To(ConsistOf(matchCPUs(cell.CPUS.CPUs)))
+			}
+		},
+			Entry("Single cell, single page, single CPU per cell", 1, 1, 1),
+			Entry("Multiple cells, multiple pages, multiple CPUs per cell", 4, 3, 4),
+		)
+	})
+})
+
+type option func(caps *libvirtxml.Caps)
+
+func newLibvirtCapsTopology(opts ...option) *libvirtxml.Caps {
+	caps := &libvirtxml.Caps{
+		Host: libvirtxml.CapsHost{
+			NUMA: &libvirtxml.CapsHostNUMATopology{},
+		},
+	}
+
+	for _, f := range opts {
+		f(caps)
+	}
+
+	return caps
+}
+
+func withCells(count int) option {
+	return func(caps *libvirtxml.Caps) {
+		caps.Host.NUMA.Cells = &libvirtxml.CapsHostNUMACells{
+			Cells: make([]libvirtxml.CapsHostNUMACell, count),
+		}
+		for i := 0; i < count; i++ {
+			caps.Host.NUMA.Cells.Cells[i] = libvirtxml.CapsHostNUMACell{
+				ID: i,
+			}
+		}
+	}
+}
+
+func withCellMemory(size uint64, unit string) option {
+	return func(caps *libvirtxml.Caps) {
+		for i := range caps.Host.NUMA.Cells.Cells {
+			caps.Host.NUMA.Cells.Cells[i].Memory = &libvirtxml.CapsHostNUMAMemory{
+				Unit: unit,
+				Size: size,
+			}
+		}
+	}
+}
+
+func withCellPages(pageCount int, count uint64, unit string) option {
```

Comment:
I think `count` is confusing. What's the difference between `count` and `pageCount`?

Reply:
`pageCount` reflects the number of page size options per cell, count is the amount of pages of each page option (e.g. 1000 pages of size 4KiB). I think I can change it to `count` and `pageAmount` respectively to conform to the established convention of count == amount of objects.